pkg/kubeconfig/create.go

```diff
@@ -56,30 +54,19 @@ func CreateKubeConfigFromServiceAccount(apiServerConfigProvider utils.ApiServerC
 // 1. from the ClusterConfig
 // 2. defaults to 6443.
 func GetApiServerURL(apiServerConfigProvider utils.ApiServerConfigProvider, clientset kubernetes.Interface) (string, error) {
-	address := apiServerConfigProvider.GetAPIServerConfig().Address
+	config := apiServerConfigProvider.GetAPIServerConfig()
+	var port string
+	address := utils.GetHost(config.Address)
```

Comment:
Would personally simplify this logic. If the address is not empty you just use that (without splitting and recomposing address and port), otherwise you call `utils.GetAPIServerAddressFromMasterNode` and use the address returned (already composed of address and port). In the end, you prepend `https://` if not already present (adding port 443 instead should not be needed, as that is the default).
Doing so, you should be also able to drop the GetAddress and GetPort methods.